DEVELOPMENT.md

```diff
@@ -47,3 +47,60 @@ The short step-by-step version is:
+          frontend automatically adds links. For example if referencing an issue/PR outside
+          this repository which has the same number like an issue in the cdt-gdb-vscode repository.
+        - Prefix issues from the sibling project `cdt-gdb-adapter` with their name if a change was
+          made in cdt-gd-vscode to resolve it.
+    - If an update of `cdt-gdb-adapter` is included, then add a section `Update to cdt-gdb-adapter vX.Y.Z`.  
+      Include the release notes as listed in [`cdt-gdb-adapter` CHANGELOG](https://github.com/eclipse-cdt-cloud/cdt-gdb-adapter/blob/main/CHANGELOG.md).  
+      Make sure to prefix issue and PR links correctly if they point to the cdt-gdb-adapter repository.
+- Update the `version` entry in [package.json](./package.json) to the new version.  
+  If the release only introduces defect fixes without significant feature changes,
+  then bump the third ("patch") version digit.  
+  Bump the second ("minor") version digit when new features have been added.  
+  Update the first ("major") version digit if there are changes that remove features
+  or significantly change existing behavior.
+
+### Start the publishing
+
+After the PR has been reviewed and merged, go to the GitHub [releases page](https://github.com/eclipse-cdt-cloud/cdt-gdb-vscode/releases):
+
+- Click `Draft a new release`.
+- Click the `Select Tag` dropdown and enter the new version in the form `vX.Y.Z`.
+- Click the `Generate release notes` button. This inserts a release name based on the
+  selected tag. And creates a list of commits since the last release as release notes
+  that are shown on GitHub.
+    - If an update of `cdt-gdb-adapter` is included, then add a section `Update to cdt-gdb-adapter vX.Y.Z`.  
+      Include the release notes as listed in [`cdt-gdb-adapter` CHANGELOG](https://github.com/eclipse-cdt-cloud/cdt-gdb-adapter/blob/main/CHANGELOG.md).  
+      Make sure to prefix issue and PR links correctly if they point to the cdt-gdb-adapter repository.
+- Select whether the release is a pre-release and/or if it is the latest release to show
+  on the GitHub repository page. Usually, no change of the defaults is required.
+- Click the `Publish release` button. This creates a new release and pushes the defined tag.
+  The tag push triggers a GitHub action which builds, tests, uploads release artifacts to GitHub,
+  and finally pushes releases to the [Open VSX Registry](https://open-vsx.org/extension/eclipse-cdt/cdt-gdb-vscode)
+  and the [VS Code Marketplace](https://marketplace.visualstudio.com/items?itemName=eclipse-cdt.cdt-gdb-vscode).
+  It may take a few minutes for this and the release's asset list to complete, and for the extensions to show up
+  on the other registries.
+
+Note: If CI should fail before the GitHub asset upload, you can either try to retrigger the failing GitHub action.
+Alternatively, you can manually remove the release and (!) the tag and retry with the same
```

Comment:
I think there is a defect in our [otterdog settings](https://otterdog.eclipse.org/projects/ecd.cdt-cloud/repos/cdt-gdb-vscode#overview) if tags can be deleted without higher level intervention (being EF IT team). Although I don't know if otterdog supports tag protection.

I think instead of recommending deleting tag, if retriggering GHA doesn't work then do a new patch version as you say a few lines below.

Reply:
Other than this quibble - I approve this.

Reply:
Thanks, removed the part and did some minor beautifications in https://github.com/eclipse-cdt-cloud/cdt-gdb-vscode/pull/182/commits/7af3d63277f3f9957053b18a18700b278b59bdba